### ISSUE TITLE
Spawn electron apps as plugins

### ISSUE THREAD

**Issue by dimitrieh** [zeit/hyper#1272]
This issue originally came from: https://github.com/zeit/hyper/issues/254

@OJFord excellently described my case:

> Just imagine: editing something in atom, you save it, and change pane to the small one on Atom's left; deploy your changes. You switch tab - boom - no editor in sight, you have a couple of panes related to wherever you just deployed your code, SSH and htop say.
> If Hyper was an Atom plugin, then there'd be no escaping Atom. Suddenly all of your shell activity is inexplicably inside a text editor. This is sort of okay, since Atom does support horizontal and vertical splits; there's no reason that any one of them need actually be editing text. There'd still be no tabbing. But it also just feels wrong.
> TL;DR:
> - You launch editors from your terminal; not the other way around.
> - Hyper in Atom would destroy tabs and (Hyper's) panes.

### Proposal

Spawn electron apps within a terminal pane

> Hyper is finally a terminal that could do more than just a hacked on implementation of code editors.. it could integrate a real code editor with all the functionality that comes with it.

This way you could use hyper's panes functionality to manage and save your sessions including dedicated apps.

and it could look like this:

<img width="1440" alt="screenshot 2016-12-19 13 17 46" src="https://cloud.githubusercontent.com/assets/429659/21312647/a746477a-c5ed-11e6-93bc-8d9655ee2623.png">

cc: @controversial @amilajack @hharnisc

**Comment by RomanADavis** [comment]
`This still sounds really cool. Is anyone working on this?
